pkg/executor/set.go

```diff
@@ -172,8 +172,11 @@ func (e *SetExecutor) setSysVariable(ctx context.Context, name string, v *expres
 			dom := domain.GetDomain(e.Ctx())
 			config.GetGlobalConfig().Instance.TiDBServiceScope = valStr
 			serverID := disttaskutil.GenerateSubtaskExecID(ctx, dom.DDL().GetID())
-			_, err = e.Ctx().(sqlexec.SQLExecutor).ExecuteInternal(ctx,
-				`replace into mysql.dist_framework_meta values(%?, %?, DEFAULT)`, serverID, valStr)
+			taskMgr, err := storage.GetTaskManager()
+			if err != nil {
+				return err
+			}
+			return taskMgr.StartManagerSession(ctx, e.Ctx(), serverID, valStr)
```

Comment:
If the sysvar is not set, the node won't be used for disttask.
Is it exepected?

Reply:
why? i just replace it from explicit sql to task_table api

Reply:
Okay, I missunderstood the code